packages/webapp/pages/notifications.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { ReactElement } from 'react';
+import classNames from 'classnames';
+import { NextSeo } from 'next-seo';
+import { useInfiniteQuery, InfiniteData, useMutation } from 'react-query';
+import {
+  NotificationsData,
+  NotificationType,
+  NOTIFICATIONS_QUERY,
+  READ_NOTIFICATIONS_MUTATION,
+} from '@dailydotdev/shared/src/graphql/notifications';
+import {
+  pageContainerClassNames,
+  pageBorders,
+} from '@dailydotdev/shared/src/components/utilities';
+import request from 'graphql-request';
+import { apiUrl } from '@dailydotdev/shared/src/lib/config';
+import NotificationItem from '@dailydotdev/shared/src/components/notifications/NotificationItem';
+import EnableNotification from '@dailydotdev/shared/src/components/notifications/EnableNotification';
+import { cloudinary } from '@dailydotdev/shared/src/lib/image';
+import {
+  getLayout,
+  getLayout as getFooterNavBarLayout,
+} from '../components/layouts/FooterNavBarLayout';
+import ProtectedPage from '../components/ProtectedPage';
+
+const hasUnread = (data: InfiniteData<NotificationsData>) =>
+  data.pages.some((page) =>
+    page.notifications.edges.some(({ node }) => !node.readAt),
+  );
+
+const Notifications = (): ReactElement => {
+  const seo = <NextSeo title="Notifications" nofollow noindex />;
+  const { mutateAsync: readNotifications } = useMutation(() =>
+    request(`${apiUrl}/graphql`, READ_NOTIFICATIONS_MUTATION),
+  );
+  const queryResult = useInfiniteQuery<NotificationsData>(
+    ['notifications'],
+    ({ pageParam }) =>
+      request(`${apiUrl}/graphql`, NOTIFICATIONS_QUERY, {
+        first: 100,
+        after: pageParam,
+      }),
+    {
+      getNextPageParam: (lastPage) =>
+        lastPage?.notifications?.pageInfo?.hasNextPage &&
+        lastPage?.notifications?.pageInfo?.endCursor,
+      onSuccess: (data) => {
+        if (hasUnread(data)) {
+          readNotifications();
+        }
+      },
+    },
+  );
+
+  const length = queryResult?.data?.pages?.length ?? 0;
+
+  return (
```

Comment:
Where do we fetch the next page?

Reply:
I have created a generic component and used it in the `UpvoterList` but I actually forgot to use it here 🤣 
This should work now, I have pushed the fix.